lineage/crossval.py

```diff
@@ -1,15 +1,64 @@
 """ Cross validation. """
+import numpy as np
 from scipy.stats import bernoulli
 from copy import deepcopy
+from .Analyze import run_Analyze_over
 
 
 def hide_observation(lineages: list, percentage: float) -> list:
     """ Taking a list of lineages and the percentage of cells want to be masked, it marks those x% negative."""
     new_lineages = deepcopy(lineages)
     for new_lineage in new_lineages:
-        for cell in new_lineage.output_lineage:
-            if bernoulli.rvs(p=percentage, size=1):
-                # negate the cell observations to mask them
-                cell.obs = [-1 * o for o in cell.obs]
+        if len(new_lineage.output_lineage) < 2:
+            continue
+        else:
+            for cell in new_lineage.output_lineage:
+                if bernoulli.rvs(p=percentage, size=1):
+                    # negate the cell observations to mask them
+                    cell.obs = np.array([-1 * o for o in cell.obs])
```

Comment:
I'm not sure you want to change the type of `cell.obs` from a list to an ndarray only when masked. However, I do agree it should be an ndarray. Perhaps you can make this `cell.obs *= -1` and then find where it's being defined as a list and convert those spots to an ndarray?

Reply:
I tried to do that, but it messed with different things. If it is okay, I will keep the cell.obs as a list everywhere.

Reply:
Yep, fine to keep it this way.